Jenkinsfile

```diff
@@ -69,17 +109,75 @@ pipeline{
 				}
 			}
 		}
-		// This stage archives all logs and database backups produced by Orthoinference. It also archives the eligible/inferred files produced by orthoinference.
-		stage('Archive logs and backups'){
+		// This stage generates the graph database using the graph-importer module, and replaces the current graph db with it.
+		stage('Post: Generate Graph Database'){
+			steps{
+				script{
+					sh "git clone https://github.com/reactome/graph-importer"
+					dir("graph-importer"){
+						sh "mvn clean compile assembly:single"
+						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+							sh "java -jar target/GraphImporter-jar-with-dependencies.jar --name ${env.RELEASE_CURRENT} --user $user --password $pass --neo4j /tmp/graph.db"
+							sh "sudo service tomcat7 stop"
+							sh "sudo service neo4j stop"
+							// This static script adjusts permissions of the graph.db folder and moves it to /var/lib/neo4j/data/databases/.
+							sh "sudo bash /var/lib/jenkins/changeGraphDatabase.sh"
+							sh "sudo service neo4j start"
+							sh "sudo service tomcat7 start"
+						}
+					}
+				}
+			}			
+		}
+		// This stage runs the graph-qa script that will be emailed to Curation.
+		stage('Post: Run graph-qa'){
+			steps{
+				script{
+					sh "git clone https://github.com/reactome/graph-qa"
+					dir("graph-qa"){
+						sh "mvn clean compile assembly:single"
+						withCredentials([usernamePassword(credentialsId: 'neo4jUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+							sh "java -jar target/graph-qa-jar-with-dependencies.jar -u $user -p  $pass --verbose"
+						}
+					}
+				}
+			}
+		}
+		// This stage emails the contents of the GraphQA_Summary_vXX.csv that is generated by graph-qa for review. 
+		stage('Post: Email graph-qa output'){
+			steps{
+				script{
+					emailext (
+						body: "Hello,\n\nThis is an automated message from Jenkins regarding an update for v${currentRelease}. The Orthoinference step has finished running. Attached to this email should be the summary report output by graph-qa. Please compare this with the graph-qa output from the previous release and confirm if they look appropriate with the developer running Release. \n\nThanks!",
+						to: '$DEFAULT_RECIPIENTS',
+						from: "${env.JENKINS_RELEASE_EMAIL}",
+						subject: "Orthoinference graph-qa for v${currentRelease}",
+						attachmentsPattern: "**/graph-qa/reports/GraphQA_Summary_v${currentRelease}.csv"
+					)
+				}
+			}
+		}
+		// All databases, logs, reports, and data files generated by this step are compressed before moving them to the Reactome S3 bucket. 
+		// All remaining files/folders are then deleted that are not a part of the release-orthoinference repository.
+		stage('Post: Archive Outputs'){
```

Comment:
There are 4 or 5 stages prefixed with "Post:" Why are they all single-step stages? Why not put multiple steps in to one "Post" stage?

Reply:
From the view of Jenkins, I like having the distinct tasks viewable from the job running window. That way if something goes wrong, its easier to troubleshoot

Reply:
So, individual steps are not as visible as stages?

Reply:
Thats right, you can see the individual steps (and the individual commands) as they are executed, but the eagle-eye view only shows the stage